core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1240,10 +1243,17 @@ class DAGScheduler(
     if (errorMessage.isEmpty) {
       logInfo("%s (%s) finished in %s s".format(stage, stage.name, serviceTime))
       stage.latestInfo.completionTime = Some(clock.getTimeMillis())
+
+      // Clear failure count for this stage, now that it's succeeded.
+      // We only limit consecutive failures of stage attempts,so that if a stage is
+      // re-used many times in a long-running job, unrelated failures don't eventually cause the
+      // stage to be aborted.
+      stage.clearFailures()
```

Comment:
This looks correct, but just so I understand, we need to do this in case this stage is resubmitted in the future for a different reason?